### PR TITLE
Fix components not changing on first clicks in Editform

### LINKED ISSUE #956: FluentCheckbox or FluentSwitch does not show correct first checked state in EditForm
# 🐛 Bug Report

The checked state does not show on first check when  `FluentCheckbox` or `FluentSwitch` is used in `EditForm`. Clicking on it 3 times will show the correct checked state.

## 💻 Repro or Code Sample

[Counter.razor.txt](https://github.com/microsoft/fluentui-blazor/files/13348267/Counter.razor.txt)

## 🤔 Expected Behavior

Show the correct checked state.

## 😯 Current Behavior

Not showing the correct checked state on first check.

## 💁 Possible Solution
Workaround: use `checked` and `onchange` instead of `@bind-Value`

`<FluentCheckbox checked="@Content.IsChecked"
        @onchange="@(() => Content.IsChecked = !Content.IsChecked)">Check test</FluentCheckbox>`

## 🌍 Your Environment

* OS & Device: Windows 10
* Browser: Microsoft Edge, Google Chrome, Mozilla FireFox, Opera
* .NET6.0 and Microsoft.Fast.Components.FluentUI Version 3.3

### LINKED ISSUE #1011: fix: Wrong position of thumb in FluentSlider after attempt to move thumb
# 🐛 Bug Report
I tried to use EditForm with several FluentSliders inside dialog. 
## 💻 Repro or Code Sample
[ThumbBugTestProject.zip](https://github.com/microsoft/fluentui-blazor/files/13448423/ThumbBugTestProject.zip)

1. Click button "Open dialog" on Home page
2. Try to move thumb of any slider inside dialog

## 🤔 Expected Behavior

Thumb must stay on the same horizontal line

## 😯 Current Behavior

Thumbs jumps to new position outside his slider

### PR DESCRIPTION
**Pull request by vnbaaij**
- Fix FluentCheckbox initial clicks in EditForm
- Fix FluentSwitch initial clicks in EditForm
- Fix #1011 in a better way
- Fix #956 

When using Slider, Checkbox or Switch in an EditForm the `checked` or `horizontal` (or `vertical`) value got removed from the class attributes because of EditForm inserting `modified`/`valid`/`invalid` in the class. With this PR the `ClassValue` is overriden from the `FluentInputBase` class and the `Value` or `Orientation`  value gets re-inserted.

Also, all tests have been updated to work with this (and some new ones were added)

### CODE REVIEW

File: tests/Core/Button/FluentButtonTests.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@ namespace Microsoft.FluentUI.AspNetCore.Components.Tests.Button;
 
 public partial class FluentButtonTests : TestContext
 {
-    private TestContext TestContext => new(); // TODO: To remove and to use the `RenderComponent` inherited method.
+    private static TestContext TestContext => new(); // TODO: To remove and to use the `RenderComponent` inherited method.
```
Comment (dvoituron) [inline tests/Core/Button/FluentButtonTests.cs:10]: Why did you added `static`?

File: src/Core/Components/Checkbox/FluentCheckbox.razor.cs [hunk 2]
```diff
@@ -10,6 +11,16 @@ public partial class FluentCheckbox : FluentInputBase<bool>
     private bool? _checkState = false;
     private const string JAVASCRIPT_FILE = "./_content/Microsoft.FluentUI.AspNetCore.Components/Components/Checkbox/FluentCheckbox.razor.js";
 
+    protected override string? ClassValue
+    {
+        get
+        {
+            return new CssBuilder(base.ClassValue)
+                .AddClass("checked", Value)
```
Comment (dvoituron) [inline src/Core/Components/Checkbox/FluentCheckbox.razor.cs:19]: Could you add the name of the "when" parameter to be clearer about the meaning of "value"?
`.AddClass("checked", when: Value)`

File: src/Core/Components/Checkbox/FluentCheckbox.razor.cs [hunk 3]
```diff
@@ -10,6 +11,16 @@ public partial class FluentCheckbox : FluentInputBase<bool>
     private bool? _checkState = false;
     private const string JAVASCRIPT_FILE = "./_content/Microsoft.FluentUI.AspNetCore.Components/Components/Checkbox/FluentCheckbox.razor.js";
 
+    protected override string? ClassValue
```
Comment (dvoituron) [inline src/Core/Components/Checkbox/FluentCheckbox.razor.cs:14]: Could you move this property to the end of the properties (after CheckStateChanged)?
We'll then have Constructors / Inject / Parameters / Public properties / Public methods / Private

See: https://dvoituron.com/naming-conventions/blazor/

File: src/Core/Components/Slider/FluentSlider.razor.cs [hunk 4]
```diff
@@ -2,17 +2,24 @@
...
 public partial class FluentSlider<TValue> : FluentInputBase<TValue>
-#if NET7_0_OR_GREATER
     where TValue : System.Numerics.INumber<TValue>
-#else
-    where TValue : IComparable, IComparable<TValue>, IConvertible, IEquatable<TValue>, IFormattable
-#endif
-
 {
+
+    protected override string? ClassValue
```
Comment (dvoituron) [inline src/Core/Components/Slider/FluentSlider.razor.cs:13]: Same... could you move this property further down?

File: src/Core/Components/Switch/FluentSwitch.razor.cs [hunk 5]
```diff
@@ -1,10 +1,21 @@
 using System.Diagnostics.CodeAnalysis;
 using Microsoft.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components.Utilities;
 
 namespace Microsoft.FluentUI.AspNetCore.Components;
 
 public partial class FluentSwitch : FluentInputBase<bool>
 {
+    protected override string? ClassValue
```
Comment (dvoituron) [inline src/Core/Components/Switch/FluentSwitch.razor.cs:9]: Same :-)

File: tests/Core/Slider/FluentSliderTests.cs [hunk 6]
```diff
@@ -6,7 +6,7 @@ namespace Microsoft.FluentUI.AspNetCore.Components.Tests.Slider;
 
 public partial class FluentSliderTests: TestContext
 {
-    private TestContext TestContext => new(); // TODO: To remove and to use the `RenderComponent` inherited method.
+    private static TestContext TestContext => new(); // TODO: To remove and to use the `RenderComponent` inherited method.
```
Comment (dvoituron) [inline tests/Core/Slider/FluentSliderTests.cs:9]: Same, why did you added `static`?

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: tests/Core/Button/FluentButtonTests.cs (hunk 1 above)
Comment (vnbaaij) [reply tests/Core/Button/FluentButtonTests.cs:10]: Was a compiler suggestion

File: src/Core/Components/Checkbox/FluentCheckbox.razor.cs (hunk 3 above)
Comment (vnbaaij) [reply src/Core/Components/Checkbox/FluentCheckbox.razor.cs:14]: ✅

File: src/Core/Components/Switch/FluentSwitch.razor.cs (hunk 5 above)
Comment (vnbaaij) [reply src/Core/Components/Switch/FluentSwitch.razor.cs:9]: ✅

File: tests/Core/Slider/FluentSliderTests.cs (hunk 6 above)
Comment (vnbaaij) [reply tests/Core/Slider/FluentSliderTests.cs:9]: Compiler suggestion

File: tests/Core/Button/FluentButtonTests.cs (hunk 1 above)
Comment (dvoituron) [reply tests/Core/Button/FluentButtonTests.cs:10]: Don't always believe VS ;-) I agree on some points, but using `static` can be dangerous in some cases. Personally, I prefer not to use it, even if it's more efficient at times.

**Review by dvoituron — APPROVED** [review]

**Pull request merged**